src/main/java/com/commercetools/sync/services/impl/ChannelServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package com.commercetools.sync.services.impl;
+
+import com.commercetools.sync.services.ChannelService;
+import io.sphere.sdk.channels.Channel;
+import io.sphere.sdk.channels.ChannelDraft;
+import io.sphere.sdk.channels.ChannelDraftBuilder;
+import io.sphere.sdk.channels.ChannelRole;
+import io.sphere.sdk.channels.commands.ChannelCreateCommand;
+import io.sphere.sdk.channels.queries.ChannelQuery;
+import io.sphere.sdk.channels.queries.ChannelQueryBuilder;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.queries.QueryExecutionUtils;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+
+public final class ChannelServiceImpl implements ChannelService {
```

Comment:
> Let's imagine 2 threads are calling fetchCachedCategoryId. For both the statement externalIdToIdCache.isEmpty() could return true and they both calls API.

But this exactly shouldn't happen with `ConcurrentHashMap`. From the [JavaDoc](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html) of it:
> Bear in mind that the results of aggregate status methods including size, isEmpty, and containsValue are typically useful only when a map is not undergoing concurrent updates in other threads.